### DISCUSSION TITLE
Feature Request: Restricting Repository Role Access to Enhance Developer vs Stakeholder Separation

### DISCUSSION THREAD

**Discussion by cilerler** [community/community#64323] [Repositories]
I'm seeking a feature that allows restriction of users with a repository role, so they can only interact with issues, discussions, projects, and the wiki. Access to code, pull-requests, packages, actions, etc., should be limited. This aims to create a distinct separation between developers and stakeholders, ensuring that code, packages, and other technical elements are not exposed to non-developers.

**Comment by rifqifarhansyah** [comment]
GitHub provides a flexible permission and access control system that allows you to customize user access to various features within a repository. To achieve the separation you're looking for between developers and stakeholders, you can use the following approaches:

1. **Repository Permissions:**
   You can set repository-level permissions to control who has access to various features. However, it's important to note that GitHub's permission system is organized around team and collaborator access to the entire repository, rather than fine-grained access control to specific features. Still, you can use this approach to manage general access:

   - Create a "Developers" team that includes users with full access to the repository, including code, pull requests, actions, etc.
   - Create a "Stakeholders" team that includes users with more limited access, such as access only to issues, discussions, projects, and the wiki.
   - Adjust the permissions for each team accordingly, giving the "Developers" team more access and the "Stakeholders" team limited access.

2. **Branch Protection Rules:**
   Although you can't directly restrict access to specific features based on repository roles, you can use branch protection rules to control who can push to certain branches. This can be used to safeguard sensitive code branches. For example:

   - Configure branch protection rules on important branches, requiring review from specific users or teams before changes can be merged.
   - Grant push and merge permissions only to your "Developers" team.

3. **Using Organizations:**
   If your repository is part of an organization, you can set up a similar role-based access approach using teams within the organization. This might provide a more structured way to manage access across multiple repositories.

4. **Third-party Integrations:**
   There are third-party tools and services that offer more granular access control options and can help you achieve the separation you're looking for. Some of these tools provide fine-grained permissions for different features, enabling you to tailor access for different roles.

Remember that while these approaches provide ways to manage access, no approach is entirely foolproof. It's important to regularly review and audit your permissions to ensure that users have the appropriate access and that no sensitive information is being exposed to unauthorized users.

Additionally, GitHub's features and interface may evolve over time, so it's a good idea to stay updated with their documentation and announcements for any changes or new features related to access control and permissions.

**Reply by cilerler** [reply]
Could you send a screenshot to confirm the creation of the""Stakeholders" team that includes users with more limited access, such as access only to issues, discussions, projects, and the wiki.` a screenshot. I want to ensure this is implemented correctly rather than just a textual explanation like ChatGPT does.
